File: app/MMarketx/js/src/data/MarketDay.js
// Copyright 09-Mar-2020 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

/**
    Market day
**/
export default class MarketDay {
  /**
      @param {string} date
      @param {number} hopen
      @param {number} mopen
      @param {number} hclose
      @param {number} mclose
   */
  constructor (
    date,
    hopen,
    mopen,
    hclose,
    mclose
  ) {
    this._date = date;
    this._hopen = hopen;
    this._mopen = mopen;
    this._hclose = hclose;
    this._mclose = mclose;
  }

  /**
      @return {string}
  **/
  get date () {
    return this._date;
  }

  /**
      @return {number}
  **/
  get hopen () {
    return this._hopen;
  }

  /**
      @return {number}
  **/
  get mopen () {
    return this._mopen;
  }

  /**
      @return {number}
  **/
  get hclose () {
    return this._hclose;
  }

  /**
      @return {number}
  **/
  get mclose () {
    return this._mclose;
  }

  /**
      @return {!Array<?>}
  **/
  toJs () {
    return [
      this._date,
      this._hopen,
      this._mopen,
      this._hclose,
      this._mclose
    ];
  }

  /**
      @param {!Array<?>} serial
      @return {!MarketDay}
  **/
  static fromJs (serial) {
    return new MarketDay(
      serial[0],
      serial[1],
      serial[2],
      serial[3],
      serial[4]
    );
  }
}

File: app/QMarket/go/data/calendar/marketDay/marketDay.go
// Copyright 16-Oct-2021 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

// Market day data.
package marketDay

import (
	"github.com/dedeme/golib/json"
)

type T struct {
	date   string
	hopen  int
	mopen  int
	hclose int
	mclose int
}

// Market day
func (tt *T) Date() string {
	return tt.date
}

// Open hour.
func (tt *T) Hopen() int {
	return tt.hopen
}

// Open minute.
func (tt *T) Mopen() int {
	return tt.mopen
}

// Close hour.
func (tt *T) Hclose() int {
	return tt.hclose
}

// Close minute.
func (tt *T) Mclose() int {
	return tt.mclose
}

func (tt *T) ToJs() json.T {
	return json.Wa([]json.T{
		json.Ws(tt.date),
		json.Wi(tt.hopen),
		json.Wi(tt.mopen),
		json.Wi(tt.hclose),
		json.Wi(tt.mclose),
	})
}

func FromJs(js json.T) *T {
	a := js.Ra()
	return &T{
		a[0].Rs(),
		a[1].Ri(),
		a[2].Ri(),
		a[3].Ri(),
		a[4].Ri(),
	}
}
